tsdb/head.go

```diff
@@ -525,27 +525,57 @@ Outer:
 					break Outer
 				}
 
-				if created {
-					// If this series gets a duplicate record, we don't restore its mmapped chunks,
-					// and instead restore everything from WAL records.
-					series.mmappedChunks = mmappedChunks[series.ref]
+				if h.lastSeriesID.Load() < s.Ref {
+					h.lastSeriesID.Store(s.Ref)
+				}
 
-					h.metrics.chunks.Add(float64(len(series.mmappedChunks)))
-					h.metrics.chunksCreated.Add(float64(len(series.mmappedChunks)))
+				if !created {
+					// There's already a different ref for this series.
+					multiRef[s.Ref] = series.ref
```

Comment:
don't we want to `continue` here? :thinking:

Reply:
If there is a different ref already, that means this was a duplicate record (hence from the future). So we need to check for m-mapped chunks from later point of time (that's exactly the enhancement here :) to replace old chunks with new ones).